src/wallet/rpcwallet.cpp

```diff
@@ -426,8 +426,6 @@ UniValue sendtoaddress(const UniValue &params, bool fHelp)
             HelpExampleRpc(
                 "sendtoaddress", "\"1M72Sfpbz1BPpXFHz9m3CdqATR44Jvaydd\", 0.1, \"donation\", \"seans outpost\""));
 
-    LOCK2(cs_main, pwalletMain->cs_wallet);
-
     CTxDestination dest = DecodeDestination(params[0].get_str());
```

Comment:
are you sure about taking these locks out...seems we would at least need cs_wallet?

Reply:
The locks are moved.
The code is read-only in the wallet until it gets to CWallet::CommitTransaction.  There I need to not have cs_main locked while doing AcceptToMemoryPool because that function now needs to let the tx admission processing threads run.  After AcceptToMemoryPool, the cs_main and cs_wallet locks are taken.

What I think may happen in the worst case is if you call sendtoaddress simultaneously from different processes, the 2 CreateTransaction calls may end up creating 2 tx that use the same inputs.  One of these will be rejected during AcceptToMemoryPool so one of the two sendtoaddress calls will fail with an exception.  However, I do not think that anyone is using the wallet with the kind of load required to trigger this exception since it can't handle that kind of load anyway.